internal/storage/integration/integration.go

```diff
@@ -229,17 +229,47 @@ func (s *StorageIntegration) testGetLargeTrace(t *testing.T) {
 		traces, err := v1adapter.V1TracesFromSeq2(iterTraces)
 		if err != nil {
 			t.Logf("Error loading large trace: %v", err)
+			return false
 		}
 		if len(traces) == 0 {
 			return false
 		}
 		actual = traces[0]
-		return err == nil && len(actual.Spans) >= len(expected.Spans)
+		return len(actual.Spans) == len(expected.Spans)
```

Comment:
keep this as `>=`. There could be failures and retries when writing spans, which could result in more than one copy of the span ending up in ES.